### PR TITLE
[DESIS Internal] SS-29874: Integrate 'Add/Remove Implicit Hydrogens' functionality to Ketcher

### PR DESCRIPTION
**Pull request by pradeepnschrodinger**
Issue: [SS-29874](https://jira.schrodinger.com/browse/SS-29874)

Added a tool button that will call RdKitJS's API to add/remove explicit hydrogens.

Primary: @apar03

### CODE REVIEW

File: src/script/ui/action/rdkitjstools.js [hunk 1]
```diff
@@ -34,5 +34,13 @@ export default {
 		},
 		className: 'rdkitjs',
 		disabled: () => true
+	},
+	explicitHydrogen: {
+		title: 'Add/Remove Explicit Hydrogen',
```
Comment (pradeepnschrodinger) [inline src/script/ui/action/rdkitjstools.js:39]: not sure if we need a shortcut field for the tool button

File: src/script/ui/state/rdkitjs/index.js [hunk 2]
```diff
@@ -90,15 +91,44 @@ function deleteAllSGroupsWithName(restruct, action, fieldName) {
...
 		default:
 			return '';
 	}
 }
+
+/**
+ * @param struct {Struct}
+ * @return {StructDetails}
+ */
+function getStructDetails(struct) {
```
Comment (pradeepnschrodinger) [inline src/script/ui/state/rdkitjs/index.js]: Helper function that stores more details about the structure, For now, it only has 1 field.

File: src/script/ui/action/rdkitjstools.js (hunk 1 above)
Comment (apar03) [reply src/script/ui/action/rdkitjstools.js:39]: I don't think we do

File: src/icons/explicitHydrogen.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="none" d="M0 0h24v24H0z" data-name="Trim Area"/><path d="M11 11H5V5H3v14h2v-6h6v6h2V5h-2v6zm8-4V5h-2v2h-2v2h2v2h2V9h2V7h-2zm-4 8h6v2h-6z" fill="#505050"/></svg>
```
Comment (apar03) [inline src/icons/explicitHydrogen.svg:1]: curious, where did you find this SVG?

File: src/script/chem/struct/index.js [hunk 4]
```diff
@@ -44,6 +44,11 @@ function Struct() {
 	this.sGroupForest = new SGroupForest(this);
 }
 
+/** @return {?Atom} */
+Struct.prototype.hasExplicitHydrogen = function () {
```
Comment (apar03) [inline src/script/chem/struct/index.js:48]: this method is going to return the first atom, that is a Hydrogen - which doesn't seem consistent with the method name. Maybe you want to return a boolean, which tells whether the molecule has explicit Hydrogen turned on? (i.e., from the code, whether the below find check is null or not)

File: src/script/ui/state/rdkitjs/index.js [hunk 5]
```diff
@@ -90,15 +91,44 @@ function deleteAllSGroupsWithName(restruct, action, fieldName) {
...
+	return {
+		hasExplicitHydrogen: !!struct.hasExplicitHydrogen(),
```
Comment (apar03) [inline src/script/ui/state/rdkitjs/index.js]: post this comment - https://github.com/schrodinger/ketcher/pull/16/files#r394796405, we could get rid of the double negation. Also, does it then seem worth extracting into a separate 'getStructDetails'? Maybe we could simply inline it there.

File: src/icons/explicitHydrogen.svg (hunk 3 above)
Comment (pradeepnschrodinger) [inline src/icons/explicitHydrogen.svg:1]: This doesn't work for some reason... Anyway I'll ask Robert to give me an SVG icon file.

File: src/script/chem/struct/index.js (hunk 4 above)
Comment (pradeepnschrodinger) [reply src/script/chem/struct/index.js:48]: agreed thanks

File: src/script/ui/state/rdkitjs/index.js (hunk 5 above)
Comment (pradeepnschrodinger) [reply src/script/ui/state/rdkitjs/index.js]: removed, we'll make it into a function if it's actually required at that point

File: src/icons/explicitHydrogen.svg (hunk 3 above)
Comment (pradeepnschrodinger) [reply src/icons/explicitHydrogen.svg:1]: strange... it works after a server restart. Anyway will wait for Robert.

**Pull request closed without merging**